File: JavaFitNesse/trunk/NCP_FitNesse/src/test/java/com/sqs/nochangeparking/fitfixtures/TheCallCentreFixture.java
package com.sqs.nochangeparking.fitfixtures;

import fit.Fixture;
import fitlibrary.DoFixture;
import fitlibrary.SetFixture;

import com.sqs.nochangeparking.core.CallCentre;

public class TheCallCentreFixture extends DoFixture{

	private CallCentre _theCallCentre;
	
	public TheCallCentreFixture() {
		_theCallCentre = CallCentre.getInstance();
		this.systemUnderTest = _theCallCentre;
	}

	public Fixture setupPayments() {
		return new SetupPaymentsFixture(_theCallCentre.getTodaysPayments());
	}
	public Fixture setupDefaults() {
		return new SetupDefaultsFixture(_theCallCentre);
	}
	
	public Fixture todaysPayments() {
		return new SetFixture(_theCallCentre.getTodaysPayments());
	}	
}

File: trunk/NoChangeParking/src/NoChangeParking.FitFixtures/TheCallCentreFixture.cs
using System;
using System.Collections.Generic;
using System.Text;
using NoChangeParking.Core;
using fit;
using fitlibrary;

namespace NoChangeParking.FitFixtures {
    public class TheCallCentreFixture:DoFixture {
		public TheCallCentreFixture() {
			mySystemUnderTest = CallCentre.Instance;
		}

		public Fixture TodaysPayments() {
			return new SetFixture(CallCentre.Instance.TodaysPayments);
		}

		private CalendarFixture myCalendar;
		public Fixture SetupCalendar() {
			myCalendar = new CalendarFixture();
			return myCalendar;
		}

		public Fixture CheckParkingValidityFor(string time) {
			DateTime inspectionTime = myCalendar.Lookup(time);
			return new CheckExpiryFixture(myCalendar, inspectionTime);
		}
	}

}
